test/mitmproxy/addons/test_export.py

```diff
@@ -93,6 +93,17 @@ def test_export(tmpdir):
         assert qr(f)
         os.unlink(f)
 
+        with mock.patch("mitmproxy.addons.export.open") as m:
+            m.side_effect = [PermissionError("Permission denied"),
+                             IsADirectoryError("Is a directory"),
+                             FileNotFoundError("No such file or directory")]
+            for i in range(3):
```

Comment:
I would prefer looping through each error individually. You could make use of a new test function and `pytest.parametrize` to set the side effect (exception), and also the expected log message.

Reply:
Thanks for the advice :) 
I just followed this comment https://github.com/mitmproxy/mitmproxy/pull/2761#issuecomment-355808629
I'll update this PR.